### ISSUE TITLE
#291 causes drastic performance drop in `DetectIdentifiers`

### ISSUE THREAD

**Issue by cpiber** [dynamicexpresso/DynamicExpresso#331]
We have around 115 equations on which we call `DetectIdentifiers`. Before (v2.16.1), this took around 50ms. After the change, it takes 17.1 *seconds*, all of which is spent in evaluating the regex (of which 50% is spent in `Match.AddMatch`).

Interestingly, #323 improves this somewhat to 16.7 seconds. In any case, a **330 times** slowdown seems a bit excessive.

**Comment by cpiber** [comment]
Turns out the `\b` is responsible, just adding that in the front drops us back down to 50ms

```diff
		private static readonly Regex RootIdentifierDetectionRegex =
-			new Regex(@"(?<id>@?[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}_]*)", RegexOptions.Compiled);
+			new Regex(@"\b(?<id>@?[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}_]*)", RegexOptions.Compiled);

		private static readonly Regex ChildIdentifierDetectionRegex = new Regex(
-			@"(?<id>@?[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}_]*(\.[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}_]*)*)",
+			@"\b(?<id>@?[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}_]*(\.[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}_]*)*)",
			RegexOptions.Compiled);
```

@metoule can you please comment in more detail why you removed it?

**Comment by metoule** [comment]
@cpiber based on my comment on #291, it was to support identifiers starting with a @ (e.g. `@this`). I'll check if it can be reintroduced

**Comment by cpiber** [comment]
@metoule I have something added in #332, please check it, seems I understood your comment right

**Comment by davideicardi** [comment]
Closed by #332
